### PR TITLE
Create NPM modules for Client and Server

### PR DESCRIPTION
**Pull request by rmi22186**

### CODE REVIEW

**Comment by rmi22186** [comment]
I'd like to get a PR for this for now before we move to lerna, to make sure everything here looks okay to you guys

**Comment by rmi22186** [comment]
this is good for review now:
* this PR turns both client and adobe into npm modules
* each of client and server now import the media heartbeat kit and loads the SDK dynamically if it's in test mode or not
  * i am temporarily checking in `AdobeHeartBeatKit.esm.js` so that the imports above work, this will be removed in favor of a new file structure for lerna which i will work on in parallel

**Comment by rmi22186** [comment]
you can ignore all the dist/ files, i'd say most important are `src/AdobeKit-dev.js` and `AdobeServerSideKit-dev.js`

once this is approved, i'm going to change the folder structure to align w lerna, and pull in https://git.corp.mparticle.com/mParticle/mparticle-web-integration-adobe-heartbeat to live in its own folder in this repo as well

File: src/AdobeServerSideKit-dev.js [hunk 1]
```diff
@@ -35,16 +41,42 @@
                 if (mcID && mcID.length > 0) {
                     setMCIDOnIntegrationAttributes(mcID);
                 }
+
+                if (forwarderSettings.mediaTrackingServer) {
+
```
Comment (alexs-mparticle) [inline src/AdobeServerSideKit-dev.js]: Nitpick. Unnecessary space.

Comment (rmi22186) [reply src/AdobeServerSideKit-dev.js]: will fix in next pr. thanks

**Pull request merged**